zhaquirks/niko/switch.py

```diff
@@ -0,0 +1,313 @@
+"""Niko switches (only Niko Double connectable switch,10A (552-721X2) at this time)"""
+
+from zigpy import types as t
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+from zigpy.zcl.clusters.general import (
+    Basic,
+    GreenPowerProxy,
+    Groups,
+    Identify,
+    OnOff,
+    Ota,
+    Scenes,
+)
+from zigpy.zcl.clusters.homeautomation import Diagnostic
+
+from zhaquirks import EventableCluster
+from zhaquirks.const import (
+    ARGS,
+    BUTTON_1,
+    BUTTON_2,
+    BUTTON_3,
+    BUTTON_4,
+    COMMAND_CLICK,
+    COMMAND_HOLD,
+    COMMAND_RELEASE,
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.niko import NIKO
+
+
+class NikoConfigCluster(CustomCluster):
+    """manufacturer specific cluster related to device settings"""
+
+    class ButtonOperationMode(t.enum8):
+        # WARN: while one can technically write 0x00 to the operationMode attribute
+        #       this seems to brick the device and it will need to be rejoined
+        Decoupled = 0x01
+        ControlRelay = 0x02
+
+    class LedState(t.uint8_t):
+        Off = 0x00
+        On = 0xFF
```

Comment:
Hmm, I'm not sure we can/should extend `t.uint8_t` like this (and "abuse" it as an enum). For example, it might break if it's used as a select config entity in HA(?), as it's not really an enum.
(But for that, it would be preferred to have an on/off switch anyway.)

~~For now, can you leave this out and just use the `t.uint8_t` type in the attribute def? (with a short comment for `Off` and `On`).~~
Actually, for now, wrap it in an `t.enum_factory` like in the first example below. It makes it a proper enum whilst keeping the "unsigned integer" "Zigbee type". Still not optimal though – it should be handled better in the future.

I do see it's already done somewhat similar in other places though:
- https://github.com/zigpy/zha-device-handlers/blob/0df6f2b6a7c3c895a73bedcc5846108c3050897b/zhaquirks/xiaomi/aqara/tvoc.py#L70-L76 (^ this is an interesting way to do it, as it avoids zigpy setting the type to "enum" instead of "int" [here](https://github.com/zigpy/zigpy/blob/6aeb8a6816585a4bda67e9ea1b125568c18fa702/zigpy/zcl/foundation.py#L215))
- https://github.com/zigpy/zha-device-handlers/blob/0df6f2b6a7c3c895a73bedcc5846108c3050897b/zhaquirks/xiaomi/aqara/opple_switch.py#L63-L67 (^ also interesting)
- https://github.com/zigpy/zha-device-handlers/blob/0df6f2b6a7c3c895a73bedcc5846108c3050897b/zhaquirks/sinope/thermostat.py#L65-L69 (^ not an Enum at all, should probably be changed)

The way we can implement this might slightly change with [quirks v2](https://github.com/zigpy/zigpy/discussions/1312#discussioncomment-8191961).
With quirks v2, we should be able to define a custom select entity for this. I'm pretty sure that would also need this class to be a "proper enum" (e.g. with the `t.enum_factory`)
(IMO, an on/off switch would be better, but I think the on/off values are hard-coded to 0x0 / 0x1 right now?)